atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1308,48 +1287,42 @@ protected ConflictHandler getConflictHandlerForTable(TableReference tableRef) {
             tableRef);
     }
 
-    private String getExpiredLocksErrorString(@Nullable LockRefreshToken commitLocksToken,
-                                              Set<LockRefreshToken> expiredLocks) {
+    private String getExpiredLocksErrorString(@Nullable LockTokenV2 commitLocksToken,
+                                              Set<LockTokenV2> expiredLocks) {
         return "The following external locks were required: " + externalLocksTokens
             + "; the following commit locks were required: " + commitLocksToken
             + "; the following locks are no longer valid: " + expiredLocks;
     }
 
-    private void throwIfExternalAndCommitLocksNotValid(@Nullable LockRefreshToken commitLocksToken) {
-        Set<LockRefreshToken> expiredLocks = refreshExternalAndCommitLocks(commitLocksToken);
+    private void throwIfPreCommitRequirementsNotMet(@Nullable LockTokenV2 commitLocksToken) {
+        Set<LockTokenV2> expiredLocks = refreshExternalAndCommitLocks(commitLocksToken);
```

Comment:
Yeah this part probably merits some explanation. So there are three types of locks that need to be refreshed here:
1) The immutable timestamp lock 
2) advisory locks (i.e., locks obtained passed to `LockAwareTransactionManager#runTaskWithLocksWithRetry`)
3) The transaction's commit locks

The locks that I refactored into the `PreCommitValidation` are (2), because those locks are from the `RemoteLockService`, and all other locks are from the `TimelockService`. It might make sense to also put (1) in the validation, though it would result in an extra roundtrip since currently we can batch the refresh.

(3) cannot be in the validation since the locks are acquired only at commit time, after the `SnapshotTransaction` has been constructed

To make this less confusing, maybe we should rename these things so they are less abstract:
`externalLocks` -> `immutableTimestampLock`
`PreCommitValidation` -> `AdvisoryLockCheck`